src/Vision/ImageAnnotatorTrait.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Vision;
+
+use google\cloud\vision\v1\FaceAnnotation;
+use google\cloud\vision\v1\Image;
+use google\cloud\vision\v1\ImageSource;
+use google\cloud\vision\v1\Position;
+use InvalidArgumentException;
+
+trait ImageAnnotatorTrait
+{
+    private static $URL_SCHEMES = [
+        'http',
+        'https',
+        'gs',
+    ];
+
+    private $client;
+
+    protected function setClient($client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @param resource|string|Image $image
+     * @param \Google\Cloud\Vision\V1\Feature\Type[] $features
+     * @param array $optionalArgs {
+     *
+     *     @type array $maxResults An array of maximum number of features to return
+     *     @type \Google\Cloud\Vision\V1\ImageContext $imageContext
+     *     @type string $requestClass The request class to use
+     *     @type string $featureClass The feature class to use
+     *     @type \Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeoutMillis is ignored.
+     *     @type int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     * @return \Google\Cloud\Vision\V1\AnnotateImageResponse The server response
+     */
+    protected function annotateImageImpl($image, $features, $optionalArgs = [])
+    {
+        $defaultArgs = [
+            'maxResults' => [],
+            'imageContext' => null,
+            'requestClass' => '\google\cloud\vision\v1\AnnotateImageRequest',
+            'featureClass' => '\google\cloud\vision\v1\Feature',
+        ];
+        $optionalArgs = $optionalArgs + $defaultArgs;
```

Comment:
It has been awhile since I have written PHP, so this is meant as a sanity check: Is this backwards? It looks backwards. I would expect values defined in `$optionalArgs` to take precedence over values in `$defautlArgs`. When you take associative arrays and run `+` on them, does the left-side argument win in a collision?